stainless_extraction/src/expr.rs

```diff
@@ -365,7 +370,25 @@ impl<'a, 'l, 'tcx> BodyExtractor<'a, 'l, 'tcx> {
         .skip(class_def.map_or(0, |cd| cd.tparams.len())),
       span,
     );
-    let args = self.extract_expr_refs(args.to_vec());
+    let mut args = self.extract_expr_refs(args.to_vec());
+
+    // If the function has trait bounds but is not on a class, we must find the
+    // corresponding evidence arguments.
+    if class_def.is_none() && !trait_bounds.is_empty() {
+      let type_substs = tparams
+        .iter()
+        .map(|st::TypeParameterDef { tp }| (*tp).into())
+        .zip(arg_tps_without_parents.clone())
```

Comment:
Can we pass a slice here instead of cloning?

Reply:
I still need a `.copied()` because otherwise the type of the elements is `&st::Type` instead of `st::Type` and collect doesn't work for the former.